lib/core/src/lib/auth/services/authentication.service.ts

```diff
@@ -30,10 +30,15 @@ import { RedirectAuthService } from '../oidc/redirect-auth.service';
 export class AuthenticationService implements AuthenticationServiceInterface, ee.Emitter {
     onLogin: Subject<any> = new Subject<any>();
     onLogout: Subject<any> = new Subject<any>();
+    onTokenReceived: Subject<any> = new Subject<any>();
 
     constructor(private injector: Injector, private redirectAuthService: RedirectAuthService) {
         this.redirectAuthService.onLogin.subscribe((value) => this.onLogin.next(value));
 
+        this.redirectAuthService.onTokenReceived.subscribe(
```

Comment:
Can you please add the unit test?

Reply:
Hey Amedeo 👋, I’m open to adding a unit test for this, as I was expecting feedback specifically on that. Considering I’m still getting familiar with ADF, could you guide me on the level at which I should introduce the unit test? Should I create a new test file (perhaps for the `RedirectAuthService`), or would it be more appropriate to implement this test at a different level?

Reply:
Hi Ricardo 👋, you could create a new test file for `RedirectAuthService` to test whenever a token is received and then test the subscription whitin the `lib/core/src/lib/auth/services/authentication.service.spec.ts`

Reply:
Ok, thank you, I'll do the test then 👋

Reply:
Added the unit test, I'll ask for a re-review 😄